src/value/ion_c_reader.rs

```diff
@@ -0,0 +1,220 @@
+use crate::result::IonResult;
```

Comment:
PR tour: the contents of this file were moved wholesale from the `value::reader` module when I introduced the `NativeElementReader`. Now the trait lives in `value::reader` while the implementations live in `value::ion_c_reader` and `value::native_reader` respectively.